### DISCUSSION TITLE
uuid support

### DISCUSSION THREAD

**Discussion by Lukerd-29-00** [sqlalchemy/sqlalchemy#8810] [Ideas]
It would be nice to have a UUID column integrated with python's uuid package, since that's a pretty common way to use primary keys. It would essentially just be equivalent to a string(16) column, but you could do something like query.filter(Table.uid==my_id) instead of query.filter(Table.uid==my_id.bytes)

**Comment by zzzeek** [comment]
a backend agnostic UUID, with a modern default of using Python uuid, is in 2.0:

https://docs.sqlalchemy.org/en/20/changelog/changelog_20.html#change-e673d427c771a23e933070c033f8353a

https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Uuid

**Reply by zzzeek** [reply]
though it stores 32-char hex string for readability.  It could have an option to store as 16 bytes instead.
